framework/test/unit/modules/interoperability/mainchain/commands/state_recovery_init.spec.ts

```diff
@@ -120,45 +134,193 @@ describe('Mainchain StateRecoveryInitCommand', () => {
 		interopStoreMock = {
 			createTerminatedStateAccount: jest.fn(),
 			hasTerminatedStateAccount: jest.fn().mockResolvedValue(true),
+			getOwnChainAccount: jest.fn(),
+			getChainAccount: jest.fn(),
 		};
 
 		jest
 			.spyOn(stateRecoveryInitCommand, 'getInteroperabilityStore' as any)
 			.mockImplementation(() => interopStoreMock);
 	});
 
-	it('should create a terminated state account when there is none', async () => {
-		// Arrange & Assign & Act
-		await stateRecoveryInitCommand.execute(commandExecuteContext);
+	describe('verify', () => {
+		beforeEach(() => {
+			mainchainAccount = {
+				name: 'mainchain',
+				networkID,
+				lastCertificate: {
+					height: 10,
+					stateRoot: getRandomBytes(32),
+					timestamp: 100 + LIVENESS_LIMIT,
+					validatorsHash: getRandomBytes(32),
+				},
+				status: CHAIN_ACTIVE,
+			};
+			const ownChainAccount = {
+				name: 'mainchain',
+				id: MAINCHAIN_ID,
+				nonce: BigInt('0'),
+			};
+			interopStoreMock = {
+				createTerminatedStateAccount: jest.fn(),
+				hasTerminatedStateAccount: jest.fn().mockResolvedValue(true),
+				getOwnChainAccount: jest.fn().mockResolvedValue(ownChainAccount),
+				getChainAccount: jest.fn(),
+			};
+			commandVerifyContext = transactionContext.createCommandVerifyContext<StateRecoveryInitParams>(
+				stateRecoveryInitParams,
+			);
+			jest.spyOn(sparseMerkleTree, 'verify').mockReturnValue(true);
+		});
 
-		const accountFromStore = await terminatedStateSubstore.getWithSchema(
-			getIDAsKeyForStore(transactionParams.chainID),
-			terminatedStateSchema,
-		);
+		it('should return status OK for valid params', async () => {
+			const result = await stateRecoveryInitCommand.verify(commandVerifyContext);
+			expect(result.status).toBe(VerifyStatus.OK);
+		});
+
+		it('should return error if chain id is same as mainchain id or own chain account id', async () => {
+			transactionParams.chainID = MAINCHAIN_ID;
+			encodedTransactionParams = codec.encode(stateRecoveryInitParams, transactionParams);
+			transaction = new Transaction({
+				moduleID: MODULE_ID_INTEROPERABILITY,
+				commandID: COMMAND_ID_STATE_RECOVERY_INIT,
+				fee: BigInt(100000000),
+				nonce: BigInt(0),
+				params: encodedTransactionParams,
+				senderPublicKey: getRandomBytes(32),
+				signatures: [],
+			});
+			transactionContext = createTransactionContext({
+				transaction,
+				stateStore,
+			});
+			commandVerifyContext = transactionContext.createCommandVerifyContext<StateRecoveryInitParams>(
```

Comment:
Instead of creating a new transaction, transaction context and verify context, you can simply modify the params directly in the context. This would reduce the time needed to execute the test suite.